backend/coreapp/views/scratch.py

```diff
@@ -517,14 +530,21 @@ def export(self, request: Request, pk: str) -> HttpResponse:
     def family(self, request: Request, pk: str) -> Response:
         scratch: Scratch = self.get_object()
 
-        if scratch.target_assembly.source_asm is None or is_asm_empty(
-            scratch.target_assembly.source_asm.data
-        ):
-            family = Scratch.objects.filter(slug=scratch.slug)
-        else:
+        if is_contentful_asm(scratch.target_assembly.source_asm):
+            assert scratch.target_assembly.source_asm is not None
+
             family = Scratch.objects.filter(
                 target_assembly__source_asm__hash=scratch.target_assembly.source_asm.hash,
             ).order_by("creation_time")
+        elif (
+            scratch.target_assembly.elf_object is not None
+            and len(scratch.target_assembly.elf_object) > 0
+        ):
+            family = Scratch.objects.filter(
+                target_assembly__hash=scratch.target_assembly.hash,
+            ).order_by("creation_time")
+        else:
+            family = Scratch.objects.filter(slug=scratch.slug)
```

Comment:
deduplicate this and create a function that returns `family`? (letting this caller add an .order_by call)